test/python/simulator/test_qasm_simulator.py

```diff
@@ -69,6 +70,59 @@ def test_qasm_simulator(circuit: QuantumCircuit, backend: QasmSimulatorBackend,
         assert abs(target[key] - counts[key]) < threshold
 
 
+def test_qasm_simulator_support_parametrized_gates(backend: QasmSimulatorBackend, shots: int):
+    """Test backend's adequate support of parametrized gates"""
+
+    theta_a = Parameter("theta_a")
+    theta_b = Parameter("theta_b")
+    theta_c = Parameter("theta_c")
+    circuit_1 = QuantumCircuit(2)
+    circuit_2 = QuantumCircuit(2)
+    circuit_1.ry(theta_a, 0)
+    circuit_1.rx(theta_b, 1)
+    circuit_2.rx(theta_c, 0)
+
+    # Test backend raises the right type of errors
+    with pytest.raises(AssertionError) as exc_info:
+        backend.run([circuit_1, circuit_2], [[np.pi / 2, np.pi / 2]], shots=shots).result()
+
+    assert str(exc_info.value) == "The number of circuits to simulate does not match the size of the parameter list."
+
+    with pytest.raises(AssertionError) as exc_info:
+        backend.run([circuit_1], [[np.pi / 2]], shots=shots).result()
+
+    assert (
+        str(exc_info.value)
+        == "The number of parameters in the circuit does not match the number of parameters provided."
+    )
+
+    # Test backend's correct functionality with multiple circuit
+    result = backend.run([circuit_1, circuit_2], [[np.pi / 2, np.pi / 2], [np.pi / 4]], shots=shots).result()
+    assert result.success
+
+    threshold = 0.04 * shots
+    average = shots / 4
+    counts_1 = result.get_counts(circuit_1)
+    counts_2 = result.get_counts(circuit_2)
+    target_1 = {
+        "0": average,
+        "11": average,
+        "1": average,
+    }
+    target_2 = {
+        "0": shots * (np.cos(np.pi / 8)) ** 2,
+        "1": shots * (np.sin(np.pi / 8)) ** 2,
+    }
+
+    for key in target_1:
+        assert key in counts_1
+        assert abs(target_1[key] - counts_1[key]) < threshold
+
+    for key in target_2:
+        assert key in counts_2
+        assert abs(target_2[key] - counts_2[key]) < threshold
```

Comment:
Could you maybe engineer an example that results in a non-flaky test that succeeds with certainty if everything works properly? Shouldn't be too hard by fine-tuning the circuits and parameters.

Reply:
Sure :). Could you elaborate on the things you want me to change from the test? Thanks!

Reply:
Yeah. Sure. I would love to have a test where the results do not depend on meeting a certain threshold.
I'd rather have a test with a circuit that results in a fixed computational basis state.
That way there is no margin for error and the test cannot randomly fail.

Reply:
Check last PR